config/environments/shared_deployment_config.rb

```diff
@@ -18,9 +18,12 @@
   # or in config/master.key. This key is used to decrypt credentials (and other encrypted files).
   # config.require_master_key = true
 
-  # Disable serving static files from the `/public` folder by default since
-  # Apache or NGINX already handles this.
+  # Serve static files from the `/public` folder based on environment settings
   config.public_file_server.enabled = ENV['RAILS_SERVE_STATIC_FILES'].present?
+  # Set cache control headers
+  config.public_file_server.headers = {
```

Comment:
For anyone curious, `rails assets:precompile` will precompile assets on one's own machine